### ISSUE TITLE
Case study: `tsconfck` vs `privatenumber`

### ISSUE THREAD

**Issue by JounQin** [import-js/eslint-import-resolver-typescript#262]
https://github.com/dominikg/tsconfck

vs

https://github.com/privatenumber/get-tsconfig

cc @privatenumber

**Comment by privatenumber** [comment]
Why was I tagged?

**Comment by JounQin** [comment]
I don't know if you know about `tsconfck`, I'm currently using `get-tsconfig` in several projects, but it seems it's missing some features different from `tsconfck`, would they be supported?

**Comment by privatenumber** [comment]
> missing some features different

What are they?

**Comment by JounQin** [comment]
> What are they?

Take `references` for example.

>> resolve "references" of solution-style tsconfig

> And why don't you open a discussion thread in get-tsconfig with a clear question instead of tagging me on multiple issues with no question?

I'm sorry if I was bothering you. I was just opening the two issue in case I could forget these tasks and tagging you in case you may want to take a look at `tsconfck`.

**Comment by privatenumber** [comment]
Feel free to submit feature requests in get-tsconfig. However, please make ensure that your suggestions are well-detailed and with context. Your thorough input will be appreciated.

And in the future, please initiate constructive discussions in the relevant repositories. Others may have similar questions, and it's more discoverable in the appropriate places. Thanks!

**Comment by SukkaW** [comment]
@JounQin Since `tsconfck` claims to be faster and smaller, there is no reason that why we should not replace the `get-tsconfig`.

In fact, I am going to replace `get-tsconfig` w/ `tsconfck` in the `rollup-plugin-swc3`.

**Comment by JounQin** [comment]
@SukkaW Thanks for your input, but `tsconfck` breaks my workflow currently described at https://github.com/dominikg/tsconfck/issues/14.

**Comment by dominikg** [comment]
jsconfig.json is supported in tsconfck v3, but you need to explicitly enable it with an option, there is no automatic/hybrid mode where it looks for tsconfig.json or jsconfig.json simultaneously <https://github.com/dominikg/tsconfck/pull/132> 

tsconfck is also used by vite, and there have been previous discussions about making it available to vite plugins to avoid multiple plugins all using it with separate caches. We should start a discussion on vite discord about that maybe?

**Comment by JounQin** [comment]
@dominikg Sorry I'm not a native English speaker, I don't quite understand what do you want to discuss about? Automatically seeking jsconfig.json when tsconfig.json not found or not specific passed as option.

**Comment by JounQin** [comment]
> @SukkaW Thanks for your input, but `tsconfck` breaks my workflow currently described at [dominikg/tsconfck#149](https://github.com/dominikg/tsconfck/issues/149). And I don't know if `jsconfig.json` is supported, see also [dominikg/tsconfck#14](https://github.com/dominikg/tsconfck/issues/14).

@SukkaW In case that my questions seem already been fixed after https://github.com/dominikg/tsconfck/releases/tag/tsconfck%403.0.1, do you want to raise to PR to replace `get-tsconfig` w/ `tsconfck`, or I'll do it when I'm free.

**Comment by SukkaW** [comment]
> > @SukkaW Thanks for your input, but `tsconfck` breaks my workflow currently described at [dominikg/tsconfck#149](https://github.com/dominikg/tsconfck/issues/149). And I don't know if `jsconfig.json` is supported, see also [dominikg/tsconfck#14](https://github.com/dominikg/tsconfck/issues/14).
> 
> @SukkaW In case that my questions seem already been fixed after https://github.com/dominikg/tsconfck/releases/tag/tsconfck%403.0.1, do you want to raise to PR to replace `get-tsconfig` w/ `tsconfck`, or I'll do it when I'm free.
> 
> See also [un-es/eslint-plugin-i#20](https://github.com/un-es/eslint-plugin-i/issues/20)

I am also kinda busy lately. I am gonna dig through the `tsconfck` a little more when I can get some time.

**Comment by xsjcTony** [comment]
> > What are they?
> 
> Take `references` for example.
> 
> > > resolve "references" of solution-style tsconfig
> 
> > And why don't you open a discussion thread in get-tsconfig with a clear question instead of tagging me on multiple issues with no question?
> 
> I'm sorry if I was bothering you. I was just opening the two issue in case I could forget these tasks and tagging you in case you may want to take a look at `tsconfck`.

Indeed! I've totally no idea before that it doesn't support `references`.
I recently changed my project to use the new vite template tsconfig style (with a root that only has `references` to other tsconfig files), and all my import plugin related resolving failed.

After wasting a lot of time digging into it, I found out that `get-tsconfig`'s `createPathsMatcher` doesn't support `references` at all, hence I'll need to pass all my sub-tsconfig file into `settings.project`🤣

Looking forward to support that soon by replacing it.

cc @SukkaW

**Comment by SukkaW** [comment]
> After wasting a lot of time digging into it, I found out that `get-tsconfig`'s `createPathsMatcher` doesn't support `references` at all, hence I'll need to pass all my sub-tsconfig file into `settings.project`🤣
> 
> Looking forward to support that soon by replacing it

`get-tsconfig` has a built-in resolver thing of sorts which is way easier to implement a resolver upon. `tsconfck` requires a lot of extra work.

**Comment by xsjcTony** [comment]
> > After wasting a lot of time digging into it, I found out that `get-tsconfig`'s `createPathsMatcher` doesn't support `references` at all, hence I'll need to pass all my sub-tsconfig file into `settings.project`🤣
> > Looking forward to support that soon by replacing it
> 
> `get-tsconfig` has a built-in resolver thing of sorts which is way easier to implement a resolver upon. `tsconfck` requires a lot of extra work.

That's sad. Yeah indeed replacing an underlying tool is a bit complex.
Maybe it's worth mentioning in the docs that if `references` is used, somehow user should explicitly list their referenced sub tsconfig files into `settings.project`?

**Comment by SukkaW** [comment]
> That's sad. Yeah indeed replacing an underlying tool is a bit complex. Maybe it's worth mentioning in the docs that if `references` is used, somehow user should explicitly list their referenced sub tsconfig files into `settings.project`?

Although `tsconfck` does support `references`, it is actually also impossible to use `tsconfck` in `eslint-import-resolver-typescript` due to the lacking of the sync API: https://github.com/dominikg/tsconfck/issues/183
